### ISSUE TITLE
Add Cycling view

### ISSUE THREAD

**Issue by MisterY** [organicmaps/organicmaps#2360]
A follow up to #1577 specific to cycling. Relatively similar to hiking, when no specific bike paths are available.
Other than that, prefer bike paths for the routes - both graphically and for route planning.

Water fountains, toilets.

**Comment by MisterY** [comment]
A good example of a cycling view: https://github.com/organicmaps/organicmaps/issues/1577#issuecomment-985776192

**Comment by elbotho** [comment]
Inspirations / References:
- **OSM**
  - [wiki: Cycle Routes Reference](https://wiki.openstreetmap.org/wiki/Cycle_routes)
  - [List of relevant tags](https://github.com/organicmaps/organicmaps/issues/1577#issuecomment-974475073), [Addition](https://github.com/organicmaps/organicmaps/issues/1577#issuecomment-974613116)
- **Open Cycle Map**
  - https://www.opencyclemap.org/
  - [Docs](https://www.opencyclemap.org/docs/)
  - https://wiki.openstreetmap.org/wiki/OpenCycleMap
- **OsmAnd**
  - https://osmand.net/docs/user/map/vector-maps/#routes
  - and maybe https://osmand.net/docs/user/map/vector-maps/#touring-view
- **CyclOSM**
  - https://www.cyclosm.org/#map=13/52.3671/4.8968/cyclosm
- **Waymarked Trails** (Bike paths)
  - https://cycling.waymarkedtrails.org/#?map=10.0/47.6806/7.6078 
- **geovelo**
  - https://github.com/organicmaps/organicmaps/issues/1577#issuecomment-985776192
  - https://geovelo.fr/de/route?bike-type=own&c=11.915811%2C45.415864&e-bike=false&z=11.49

**Comment by elbotho** [comment]
For me an overlay with the cycle routes would be a valuable addition (like you can switch on in OsmAnd).
Everything else is a welcome bonus of course.

**Comment by biodranik** [comment]
OSMand shows OSM routes specifically tagged for cycling, right?

**Comment by elbotho** [comment]
> OSMand shows OSM routes specifically tagged for cycling, right?

Yes, pretty sure thats how they do it. You can switch on different route overlays:

<img src="https://user-images.githubusercontent.com/1258870/189480560-fa5a8a03-0611-42dd-8a9b-b54ec7df4fdb.png" width="200" />

Fitness trails just use `route=fitness_trail` for example ([Source](https://osmand.net/blog/osmand-android-4-2-released#new-types-of-routes)). ~Did not find the tags they use for~ Here is the actuall mapping in the source:
[Did not find the tags they use for ](https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd-java/src/main/java/net/osmand/osm/RouteActivityType.java#L121-L150)

**Comment by audkar** [comment]
Please reevaluate the priority of this feature. Cities are separating more and more bike-only paths from non-bike lanes. And some routes are not valid depending on whether you are walking or biking.

**Comment by manouche** [comment]
> OSMand shows OSM routes specifically tagged for cycling, right?

yes they show it but it's so so small, alsmost useless :(

**Comment by elbotho** [comment]
> > OSMand shows OSM routes specifically tagged for cycling, right?
> 
> yes they show it but it's so so small, alsmost useless :(

I find it very useful. Did you activate the cycling routes overlay?

<img alt="OSMAnd cycling routes" src="https://user-images.githubusercontent.com/5942222/42843399-82ce4dc0-89ff-11e8-9ebc-3619f520e8f5.png">

**Comment by biodranik** [comment]
@pastk idea: add layers with routes above the map into Bookmarks and Tracks as separate items/lists where user can toggle their visibility in the same way as toggling bookmarks/tracks list visibility.

**Comment by manouche** [comment]
@elbotho Yes I activate it, let see how it look on my phone below. On the first image it is the small path in pink in the middle of the map.

![IMG_8684](https://github.com/organicmaps/organicmaps/assets/131271/4ac2922a-e3d5-46c5-8267-52f36dfc665b)
![IMG_8686](https://github.com/organicmaps/organicmaps/assets/131271/6c169ae2-1b13-44b9-87a2-b8de53a02b3a)
![IMG_8682](https://github.com/organicmaps/organicmaps/assets/131271/64af3341-68be-44c0-8373-7cafc7a0b767)

**Comment by manouche** [comment]
With cycleOSM on the same app, it is usefull :)
![IMG_8688](https://github.com/organicmaps/organicmaps/assets/131271/c3ab6388-4f30-4ca9-b3f0-77e13a5f1552)
![IMG_8687](https://github.com/organicmaps/organicmaps/assets/131271/e884d530-9201-421b-b68d-bb9d8c09b4d5)

**Comment by pgr0ss** [comment]
As another example, I like how [Mapy.cz](https://apps.apple.com/us/app/mapy-cz-navigation-maps/id411411020) shows cycle routes (note the pink solid and dotted lines):

Organic Maps             |  [Mapy.cz](https://apps.apple.com/us/app/mapy-cz-navigation-maps/id411411020)
:-------------------------:|:-------------------------:
![IMG_0153](https://github.com/organicmaps/organicmaps/assets/12998/e5bf13d7-152c-4fbd-9440-65be46758ee3)  |  ![IMG_0152](https://github.com/organicmaps/organicmaps/assets/12998/abd2bef4-e0f4-483d-be1a-12f47d825574)

**Comment by Coehill** [comment]
I really like www.cyclosm.org and use it as my base map in the OsmAND app exclusively while bicycling. Cyclosm is already used in 4 other smartphone apps ([see repo for details](https://github.com/cyclosm/cyclosm-cartocss-style)) and being open source, I think this makes the most sense.

**Comment by pastk** [comment]
- #7071

**Comment by biodranik** [comment]
User asks to show start/finish points and directions for loop/MTB tracks.

**Comment by pktiuk** [comment]
I also have some additional notes:

- Cycling maps apart from bike lanes should also show bikre-related POI like rental bike stations (like nextbike, verutilo etc), public [bike repair stations](https://wiki.openstreetmap.org/wiki/Pl:Tag:amenity=bicycle%20repair%20station), places to park bikes, etc.
- Cycling map should also show different kinds of bika paths like:
  - Regular separated, independent bike paths (like the ones going through parks etc)
  - bike lanes along regular streets
  - one-way local roads allowing two two-way traffic only for bikes

Good example is a polish app called [miejski.bike](https://play.google.com/store/apps/details?id=pl.wtopolski.android.warsawbikepath)
![obraz](https://github.com/user-attachments/assets/18b80bfb-d3fd-44dd-89f7-3b50b48b759f)

**Comment by Coehill** [comment]
> Cycling map style:
> <img alt="" width="350px" src="https://private-user-images.githubusercontent.com/1799054/361252254-ad0c0a13-cf59-48b1-b148-984399cbafd2.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2ODAyNzMsIm5iZiI6MTcyNDY3OTk3MywicGF0aCI6Ii8xNzk5MDU0LzM2MTI1MjI1NC1hZDBjMGExMy1jZjU5LTQ4YjEtYjE0OC05ODQzOTljYmFmZDIuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjZUMTM0NjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTU2ODg1YWNkODE2MWQyYWM2ZDllZjcyODU0YTI1YWU3NDQ0NjYxZjA2YTc0NTM2ZmU4NjVlODVhYmVkZTkwOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wNvy7xmPnoKgf_TVtmf4VYOKY_VzDhC0zKgOJlDoxqg"> <img alt="" width="350px" src="https://private-user-images.githubusercontent.com/1799054/361252262-f5ea7a6b-7493-4bac-a8d5-378f2d65396d.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2ODAyNzMsIm5iZiI6MTcyNDY3OTk3MywicGF0aCI6Ii8xNzk5MDU0LzM2MTI1MjI2Mi1mNWVhN2E2Yi03NDkzLTRiYWMtYThkNS0zNzhmMmQ2NTM5NmQuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjZUMTM0NjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjM0MDQ3MzlhNWU4NDlmNDVjM2ZjOGM0YTJiNDU4NzZmNTNiYTdkZTBiZDc5ODIwMDljNTk4Njk2NDAyY2Y0MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.b7y4vJTlZ1VD1_FEFcFtBDy-DtuWEX1xSo0OMF2JpzI">

The map style's bold colour for standard bike lanes is nice, if not differenr than other options. Only issue is Google has limited displaying of infrastructure, for instance, signef bike routes are not displayed. Imo cyclosm still has an advantage here. Plus, it's open source whereas there is no way to use Google's proprietary view.

**Comment by zyphlar** [comment]
See also:
https://github.com/organicmaps/organicmaps/pull/7255
https://github.com/organicmaps/organicmaps/pull/7810
https://github.com/organicmaps/organicmaps/pull/7925
https://github.com/organicmaps/organicmaps/issues/5289
https://github.com/organicmaps/organicmaps/issues/3246

**Comment by pastk** [comment]
Also need to add [`cyclestreet=yes`](https://wiki.openstreetmap.org/wiki/Key:cyclestreet) support.

**Comment by pktiuk** [comment]
@cyberchrime 

I think a lot of work is already done. You could look how other map styles (like tourism which also shows some of the amenities you mentioned) were implemented and just add similar simple style (to extend it later). Implementing **simple** (without some of the features you mentioned) cycling map style would be doable in 13-18 hours (I assume you are not very familiar with the codebase and need some time to figure out your development environment and you know the languages this repo is coded with).

> Therefore, I was wondering how much time you expect a student to need to imple ment a dedicated cycling map style (approximation in hours) with the features I provided

With all of these features it could take 30-45 hours (it is a bit of a wild guess).

But figuring out how it is supposed to look to be readable may take more time. You should also remember about review process and appying suggestions and fixes from maintainers before merging.

> Maybe a combination with a GSoC project is possible, too.

This seems to be a very good idea. AFAIK Orgainc Maps already participated in GSoC

**Comment by michaelblyons** [comment]
There is a third-party plugin for OsmAnd with [cycle-specific styles](https://osmand-rendering.github.io/CycloRoute/index_EN.html) for their vector maps.
